test/gtest/ucp/test_ucp_tag.cc

```diff
@@ -178,153 +178,158 @@ int test_ucp_tag::get_worker_index(int buf_index)
+            return NULL;
+        }
 
-    req = (request*)ucp_tag_send_nb(sender().ep(worker_index), buffer, count, datatype,
-                                    tag, send_callback);
-    if (UCS_PTR_IS_ERR(req)) {
-        ASSERT_UCS_OK(UCS_PTR_STATUS(req));
+        if (UCS_PTR_IS_ERR(req)) {
+            ASSERT_UCS_OK(UCS_PTR_STATUS(req));
+        }
+        break;
+    case SEND_NBR:
+        req = request_alloc();
+        status = ucp_tag_send_nbr(sender.ep(worker_index), buffer, count,
+                                  datatype, tag, req);
+        ASSERT_UCS_OK_OR_INPROGRESS(status);
+        if (status == UCS_OK) {
+            request_free(req);
+            return (request*)UCS_STATUS_PTR(UCS_OK);
+        }
+        break;
+    case SEND_SYNC_NB:
+        return (request*)ucp_tag_send_sync_nb(sender.ep(worker_index), buffer,
+                                              count, datatype, tag, send_callback);
+    default:
+        return NULL;
     }
+
     return req;
 }
 
 test_ucp_tag::request *
-test_ucp_tag::send_nbr(const void *buffer, size_t count,
-                            ucp_datatype_t datatype,
-                            ucp_tag_t tag, int buf_index)
+test_ucp_tag::send_nb(const void *buffer, size_t count, ucp_datatype_t datatype,
+                      ucp_tag_t tag, int buf_index)
 {
-    int worker_index = get_worker_index(buf_index);
-    ucs_status_t status;
-    request *req;
-
-    req = request_alloc();
-
-    status = ucp_tag_send_nbr(sender().ep(worker_index), buffer, count, datatype,
-                              tag, req);
+    return send(sender(), SEND_NB, buffer, count, datatype, tag, buf_index);
+}
 
-    ASSERT_UCS_OK_OR_INPROGRESS(status);
-    if (status == UCS_OK) {
-        request_free(req);
-        return (request *)UCS_STATUS_PTR(UCS_OK);
-    }
-    return req;
+test_ucp_tag::request *
+test_ucp_tag::send_nbr(const void *buffer, size_t count,
+                       ucp_datatype_t datatype,
+                       ucp_tag_t tag, int buf_index)
+{
+    return send(sender(), SEND_NBR, buffer, count, datatype, tag, buf_index);
 }
 
 
 void test_ucp_tag::send_b(const void *buffer, size_t count, ucp_datatype_t datatype,
                           ucp_tag_t tag, int buf_index)
 {
-    request *req = send_nb(buffer, count, datatype, tag, buf_index);
-
-    if (req != NULL) {
-        wait(req, get_worker_index(buf_index));
-        request_release(req);
-    }
+    send(sender(), SEND_B, buffer, count, datatype, tag, buf_index);
 }
 
 test_ucp_tag::request *
 test_ucp_tag::send_sync_nb(const void *buffer, size_t count, ucp_datatype_t datatype,
                            ucp_tag_t tag, int buf_index)
 {
-    int worker_index = get_worker_index(buf_index);
-
-    return (request*)ucp_tag_send_sync_nb(sender().ep(worker_index), buffer, count,
-                                          datatype, tag, send_callback);
+    return send(sender(), SEND_SYNC_NB, buffer, count, datatype, tag, buf_index);
```

Comment:
what is the point of common function if we have this wrapper anyway?

Reply:
the main intent of having a common function is to be able to send a data using any UCP entity (not only `sender()`)
also. the same is applicable for `recv()` common function

Reply:
`test_ucp_tag`'s functions (`send_nb`/`send_b`/`send_sync_nb` and `recv_nb`/`recv_b`) as wrappers for common ones were left in order to not change every place where they used